### PR TITLE
Fixed coverity issues in samples

### PR DESCRIPTION
**Pull request by antonrom23**
### Tickets:
 - 74762
 - 74764
 - 79236
 - 79254
 - 79464

### CODE REVIEW

File: samples/c/common/opencv_c_wrapper/bmp_reader.c [hunk 1]
```diff
@@ -70,12 +70,14 @@ int readBmpImage(const char* fileName, BitMap* image) {
 
     if (image->infoHeader.bits != 24) {
         printf("[BMP] 24bpp only supported. But input has: %d\n", image->infoHeader.bits);
-        return 3;
+        status = 3;
```
Comment (mgrdv) [inline samples/c/common/opencv_c_wrapper/bmp_reader.c:73]: Don't we want to have enum (with 3,4,5, etc. values) for code of status?

Comment (antonrom23) [reply samples/c/common/opencv_c_wrapper/bmp_reader.c:73]: I think no, It doesn't make sense and makes the code more difficult. We have output message for it.

**Review by fzhar — CHANGES_REQUESTED** [review]

**Review by vladimir-dudnik — APPROVED** [review]

File: samples/c/common/opencv_c_wrapper/bmp_reader.c [hunk 2]
```diff
@@ -86,42 +85,31 @@ int readBmpImage(const char* fileName, BitMap* image) {
...
-Exit:
-    if (0 != status && NULL != image && NULL != image->data) {
-        free(image->data);
-    }
-
-    if (NULL != input) {
-        fclose(input);
-    }
-
     return status;
```
[the remark this answers is not in the document]
Comment (antonrom23) [reply samples/c/common/opencv_c_wrapper/bmp_reader.c]: Done

File: samples/c/common/opencv_c_wrapper/bmp_reader.c [hunk 3]
```diff
@@ -70,12 +70,14 @@ int readBmpImage(const char* fileName, BitMap* image) {
 
     if (image->infoHeader.bits != 24) {
         printf("[BMP] 24bpp only supported. But input has: %d\n", image->infoHeader.bits);
-        return 3;
+        status = 3;
+        goto Exit;
```
[the remark this answers is not in the document]
Comment (antonrom23) [reply samples/c/common/opencv_c_wrapper/bmp_reader.c]: Was changed.

**Review by fzhar — APPROVED** [review]

**Pull request merged**